python/cugraph/cugraph/link_prediction/sorensen.py

```diff
@@ -0,0 +1,155 @@
+    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G.from_cudf_edgelist(gdf, source='0', destination='1')
+    >>> df = cugraph.sorensen(G)
+    """
+    if type(input_graph) is not Graph:
+        raise Exception("input graph must be undirected")
+
+    if type(vertex_pair) == cudf.DataFrame:
+        vertex_pair = renumber_vertex_pair(input_graph, vertex_pair)
+    elif vertex_pair is None:
+        pass
+    else:
+        raise ValueError("vertex_pair must be a cudf dataframe")
+
+    df = jaccard_wrapper.jaccard(input_graph, None, vertex_pair)
+    df.jaccard_coeff = ((2*df.jaccard_coeff)/(1+df.jaccard_coeff))
+    df.rename(
+        {'jaccard_coeff': 'sorensen_coeff'}, axis=1, inplace=True)
+    if input_graph.renumbered:
+        df = input_graph.unrenumber(df, "source")
+        df = input_graph.unrenumber(df, "destination")
+
+    return df
+
+
+def sorensen_coefficient(G, ebunch=None):
+    """
+
+    Parameters
+    ----------
+    graph : cugraph.Graph
+        cuGraph graph descriptor, should contain the connectivity information
+        as an edge list (edge weights are not used for this algorithm). The
+        graph should be undirected where an undirected edge is represented by a
+        directed edge in both direction. The adjacency list will be computed if
+        not already present.
+    ebunch : cudf.DataFrame
+        A GPU dataframe consisting of two columns representing pairs of
+        vertices. If provided, the sorensen coefficient is computed for the
+        given vertex pairs.  If the vertex_pair is not provided then the
+        current implementation computes the sorensen coefficient for all
+        adjacent vertices in the graph.
+
+    Returns
+    -------
+    df  : cudf.DataFrame
+        GPU data frame of size E (the default) or the size of the given pairs
+        (first, second) containing the Sorensen weights. The ordering is
+        relative to the adjacency list, or that given by the specified vertex
+        pairs.
+
+        df['source'] : cudf.Series
+            The source vertex ID (will be identical to first if specified)
+        df['destination'] : cudf.Series
+            The destination vertex ID (will be identical to second if
+            specified)
+        df['sorensen_coeff'] : cudf.Series
+            The computed sorensen coefficient between the source and
+            destination vertices
+
+    Examples
+    --------
+    >>> gdf = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G.from_cudf_edgelist(gdf, source='0', destination='1')
+    >>> df = cugraph.sorensen_coefficient(G)
+    """
+    vertex_pair = None
+
+    G, isNx = check_nx_graph(G)
+
+    if isNx is True and ebunch is not None:
+        vertex_pair = cudf.from_pandas(pd.DataFrame(ebunch))
```

Comment:
This conversion seems odd.  Can you just pass the ebunch directly to cudf?

Reply:
That conversion is indeed odd. My Cugraph `Sorensen` implementation mirrors the already existing `Jaccard `one which has a lot a dubious calls and I didn't pay attention to. I tried to make Sorensen consistent with Jaccard. I am refactoring Jaccard, overlap as well.